### ISSUE TITLE
Failed to start gauge API: Time out connecting to dotnet

### ISSUE THREAD

**Issue by RemiHoston** [getgauge/gauge#2628]
**Is your feature request related to a problem? Please describe.**
My gauge version is:
 Gauge version: 1.6.9
Plugins
-------
 csharp (0.10.6)
 dotnet (0.7.2)
 html-report (4.3.1)
 screenshot (0.3.0)
 xml-report (0.5.1)
Have found this issue occurred very frequently when running on local machine and pipeline agent. When it happened locally I can close the process from the Task Manager, but on pipeline have to retry after a few minutes. So I realized it's may a feature defect in the gauge main logic.

**Describe the solution you'd like**

It's better to close all the dotnet process when the gauge run command finished.

**Describe alternatives you've considered**

**Additional context**

**Comment by chadlwilson** [comment]
It sounds like you have diagnosed the problem as a dead or stuck runner.

Do you have a way to reproduce this? Any logs to help debug what might have happened? What environments(s)/OSes are you running from?

**Comment by RemiHoston** [comment]
Thank you for your attention, in my issues the only useful information is like:
----------------------------------------------------------------------------------------------------------------
2024-10-17T08:56:51.9589391Z MSBuild version 17.9.8+610b4d3b5 for .NET
2024-10-17T08:57:22.5882646Z Error ----------------------------------
2024-10-17T08:57:22.5882913Z 
2024-10-17T08:57:22.5883196Z [Gauge]
2024-10-17T08:57:22.5883618Z Failed to start gauge API: Timed out connecting to dotnet
2024-10-17T08:57:22.5886914Z 
2024-10-17T08:57:24.1956320Z Get Support ----------------------------
2024-10-17T08:57:24.1962697Z 
2024-10-17T08:57:24.2209038Z 	Docs:          https://docs.gauge.org
2024-10-17T08:57:24.2214110Z 	Bugs:          https://github.com/getgauge/gauge/issues
2024-10-17T08:57:24.2218619Z 	Chat:          https://github.com/getgauge/gauge/discussions
2024-10-17T08:57:24.2219847Z 
2024-10-17T08:57:24.2221192Z Your Environment Information -----------
2024-10-17T08:57:24.2222946Z 	windows, 1.6.9, aff43ef
2024-10-17T08:57:24.2224027Z 	csharp (0.10.6), dotnet (0.7.2), html-report (4.3.1), screenshot (0.3.0), xml-report (0.5.1)
-------------------------------------------------------------------------------------------------------------------------------------
It happens when after finishing a lot of cases are finished running, and a new task to run a new round gauge running. It can be solved by retry several times on the agent.

------------------------------------------ Agent info ----------------------
 |Agent.OS | Windows_NT |  
 | Agent.OSArchitecture | X64 |  
 | Agent.OSVersion | 10.0.17763
------------------------------------------------------------------------------
Hope the Gauge tool can be better and better!

**Comment by AlbertZhang6** [comment]
I met the same issue with RemiHoston, Do you have any idea? @chadlwilson

**Comment by chadlwilson** [comment]
No, unfortunately not - as I don't have any information to replicate it, proper logs from the Gauge itself (`logs/gauge.log` or similar), or any details from people about what changed when this started happening.

It seems similar to https://github.com/getgauge/gauge-dotnet/issues/196 (which supposedly was fixed in https://github.com/getgauge/gauge-dotnet/issues/197) but may have resurfaced as noted in https://github.com/getgauge/gauge-dotnet/issues/204 and https://github.com/getgauge/gauge-dotnet/issues/199

You could try rolling back to gauge-dotnet `0.5.8`. if the problem goes away or you're using async methods, it's likely related to those I guess?

**Comment by RemiHoston** [comment]
Yea, this is depend on the latest changes about `Gauge.SCharp.Lib`. Currently, we have referenced the latest version `0.11.3`, the plugin for dotnet min version is 0.7.1. It contains a break change.

**Comment by chadlwilson** [comment]
@RemiHoston Sorry, I don't understand what you are saying. You can roll back the CSharp and/or Gauge version too, right?

**Comment by RemiHoston** [comment]
Sorry, my bad. I mean the component have referenced in my project is `Gauge.CSharp.Lib 0.11.3`, which contains break changes in different the use of SuiteDataStore, SituationDataStore and SpecDataStore.
And the latest version of `Gauge.CSharp.Lib 0.11.3` requires the dotnet plugin at least `0.7.1` (if I am correct), so the dotnet 0.5.8 could not run successfully.
By the way, what's special that the latest version of dotnet 0.7.2? Could we optimize it?
![image](https://github.com/user-attachments/assets/547f6bdb-a741-409c-9c42-3e7b2d666d95)
And the if the plugin dotnet is 0.7.2, it can run smoothly:
![image](https://github.com/user-attachments/assets/d9766104-b4b1-4e27-b305-07e362a184aa)
Since upgrade `Gauge.CSharp.Lib` from 0.10.3 to 0.11.3 we have done a lot of changes in our gauge test project, and we hope can use the latest Gauge version.

**Comment by chadlwilson** [comment]
Let me try asking another way.

What was the **last reliable combination** of gauge/gauge-dotnet/`Gauge.CSharp.Lib` versions in your environment, that **did not** have this problem?

We need to narrow down when a problem started or we will be going around in circles forever guessing. Try to make it easy for maintainers, rather than require them to guess what you have done or changed in your environment or how you might use a piece of software.

**Comment by RemiHoston** [comment]
Thank you, have checked our history calls, however this version had been used for a very long time.

Gauge.CSharp.Lib.0.7.2

 Gauge version: 1.0.8
Commit Hash: 28617ea

Plugins
 -------
csharp (0.10.6)
dotnet (0.1.7)
html-report (4.0.12)
screenshot (0.0.1)
xml-report (0.2.3)

**Comment by chadlwilson** [comment]
Unfortunately it can happen for different reasons depending on what type of specs impls you are writing.

Do you have logs from gauge itself (not from your spec run) you can share?
